### PR TITLE
Added api for volume mapping deletion

### PR DESCRIPTION
**Pull request by IlyaKessel**
Dependency:

- [x] ManageIQ/manageiq#21432 

Added api for volume mapping deletion
image

### CODE REVIEW

**Comment by ereztheodorou** [comment]
@miq-bot add_label morphy/yes?

**Comment by ereztheodorou** [comment]
@chessbyte Tests seem to fail on an unrelated issue of the test environment: 

 "Cannot connect to the database!"

**Comment by agrare** [comment]
@ereztheodorou test failures look legitimate: https://app.travis-ci.com/github/ManageIQ/manageiq-api/jobs/537517629#L2079-L2082
```
  1) API configuration (config/api.yml) collections identifiers contains only valid MiqProductFeature identifiers

     Failure/Error: expect(invalid_features).to be_empty

       expected `#<Set: {"volume_mapping_delete"}>.empty?` to be truthy, got false

     # ./spec/lib/api/api_config_spec.rb:75:in `block (4 levels) in <top (required)>'
```

**Comment by agrare** [comment]
@ereztheodorou if this PR has a dependent PR (which it looks like https://github.com/ManageIQ/manageiq/pull/21432 is) then please indicate that dependency in the description.  Not just a list of related PRs but actually indicate what is dependent on what.

**Comment by agrare** [comment]
@miq-bot cross-repo-tests https://github.com/ManageIQ/manageiq/pull/21432

**Comment by IlyaKessel** [comment]
@agrare, after  https://github.com/ManageIQ/manageiq/pull/21432 merge the:
```
expected `#<Set: {"volume_mapping_delete"}>.empty?` to be truthy, got false
``` 
is fixed

**Comment by agrare** [comment]
@IlyaKessel can you write a spec test for this new action?

File: spec/requests/volume_mappings_spec.rb [hunk 1]
```diff
@@ -54,6 +54,51 @@
     end
   end
 
+  it "can delete a single volume mapiing" do
+    volume_mapping = FactoryBot.create(:volume_mapping)
+
+    api_basic_authorize("volume_mapping_delete")
```
Comment (agrare) [inline spec/requests/volume_mappings_spec.rb:60]: Shouldn't this be using `collection_action_identifier` rather than the hard-coded feature?

**Review by agrare — APPROVED** [review]
:+1: LGTM

**Comment by agrare** [comment]
@IlyaKessel looks like you just need to fix the `supports?(:delete)` failure in the specs

**Comment by IlyaKessel** [comment]
@agrare why do I have this failure?
we have supports: delete in the provider code https://github.com/ManageIQ/manageiq-providers-autosde/pull/85/files#diff-5e8790bdf8780375ca12e18360fa2543e231de5a3e3acf8b0d80136c90e15d3fR4

and it's merged.

**Comment by chessbyte** [comment]
@IlyaKessel maybe similar to https://github.com/ManageIQ/manageiq-api/pull/1080/files#diff-9ae8ab3a7672280dacf0a8f766446c141107a632451afa8303ce7e0e1b943ca2R111-114

**Comment by agrare** [comment]
@IlyaKessel because `FactoryBot.create(:volume_mapping)` builds a base `VolumeMapping` instance not the provider subclass.

**Comment by agrare** [comment]
@IlyaKessel @chessbyte is right it is better to stub the `supports?` call rather than hard-code specific provider plugin class names in the API repo, prefer https://github.com/ManageIQ/manageiq-api/pull/1075#issuecomment-930261332

File: spec/requests/volume_mappings_spec.rb [hunk 2]
```diff
@@ -69,6 +69,64 @@
...
+      expect_single_action_result(:success => true, :message => /Deleting Volume Mapping id:#{volume_mapping.id}/i, :href => api_volume_mapping_url(nil, volume_mapping))
+    end
+
+    it "Delete of multiple Volume Mappings" do
+      provider = FactoryBot.create(:ems_autosde, :name => 'Autosde')
+      volume_mapping = FactoryBot.create(:volume_mapping, :ext_management_system => provider)
+      volume_mapping_two = FactoryBot.create(:volume_mapping, :ext_management_system => provider)
+      api_basic_authorize collection_action_identifier(:volume_mappings, :delete, :post)
+
+      expect_any_instance_of(VolumeMapping).to receive(:supports?).with(:delete).and_return(true)
```
Comment (agrare) [inline spec/requests/volume_mappings_spec.rb:108]:
```suggestion
      allow_any_instance_of(VolumeMapping).to receive(:supports?).with(:delete).and_return(true)
```

**Comment by agrare** [comment]
All travis builds are failing right now but this is green locally

**Comment by ereztheodorou** [comment]
@agrare Yeah. Travis is complaining about a certificate issue. Does this mean we can merge this PR?

**Comment by agrare** [comment]
It is a general travis issue with their default xenial images, https://gitter.im/ManageIQ/manageiq/core?at=6155ce7ded08d212af4e83b5

We'll have a fix shortly and I'd prefer to merge on green travis rather than just locally to remove any other variables

**Comment by agrare** [comment]
I'm aware :wink: it'll be alright @ereztheodorou

**Comment by Fryguy** [comment]
Backported to `morphy` in commit 7b3d7af0c04516e1d9aba5268e877531f7a97b70.

```text
commit 7b3d7af0c04516e1d9aba5268e877531f7a97b70
Author: Adam Grare <adam@grare.com>
Date:   Sun Oct 3 14:10:27 2021 -0400

    Merge pull request #1075 from Autosde/volume_mapping_delete
    
    Added api for volume mapping deletion
    
    (cherry picked from commit 19b88f5419da12a4115a1b5c369aede8b129d111)
```

**Pull request merged**